includes/events/class-omise-event-charge-capture.php

```diff
@@ -1,82 +1,39 @@
 <?php
-defined( 'ABSPATH' ) or die( 'No direct script access allowed.' );
 
-if ( class_exists( 'Omise_Event_Charge_Capture' ) ) {
-	return;
-}
-
-class Omise_Event_Charge_Capture {
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * There are several cases that can trigger the 'charge.capture' event.
+ *
+ * =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+ * Credit Card
+ * charge data in payload will be:
+ *     [status: 'successful'], [authorized: 'true'], [paid: 'true']
+ *
+ */
+class Omise_Event_Charge_Capture extends Omise_Event {
 	/**
 	 * @var string  of an event name.
 	 */
-	public $event = 'charge.capture';
+	const EVENT_NAME = 'charge.capture';
 
 	/**
-	 * There are several cases that can trigger the 'charge.capture' event.
-	 *
-	 * =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-	 * Credit Card
-	 * charge data in payload will be:
-	 *     [status: 'successful'], [authorized: 'true'], [paid: 'true']
-	 *
-	 * @param  mixed $data
-	 *
-	 * @return void
+	 * @inheritdoc
 	 */
-	public function handle( $data ) {
-		if ( 'charge' !== $data->object || ! isset( $data->metadata->order_id ) ) {
-			return;
+	public function validate() {
+		if ( 'charge' !== $this->data['object'] || ! isset( $this->data['metadata']['order_id'] ) ) {
+			return false;
 		}
 
-		if ( ! $order = wc_get_order( $data->metadata->order_id ) ) {
-			return;
+		if ( ! $this->order = wc_get_order( $this->data['metadata']['order_id'] ) ) {
+			return false;
 		}
 
-		$order->add_order_note(
-			__(
-				'Omise: an event charge.capture has been caught (webhook).',
-				'omise'
-			)
-		);
-
-		switch ($data->status) {
-			case 'successful':
-				if ( $data->authorized && $data->paid ) {
-					$order->add_order_note(
-						sprintf(
-							wp_kses(
```

Comment:
This order note will be missing on Manual Capture.

Reply:
That is good point! 
I completely forgot that this class was designed to catch an event from manual capture that is being done manually from Omise Dashboard. Code pushed.